integrationtests/tests/tests.rs

```diff
@@ -1199,3 +1164,38 @@ async fn verifies_client_configs() -> Result<()> {
     })
     .await
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn cannot_replay_transactions() -> Result<()> {
+    test(4, |fed, user, bitcoin, _, _| async move {
+        fed.mine_and_mint(&user, &*bitcoin, sats(5000)).await;
+
+        let notes = user.client.notes().await;
+        let mut builder = TransactionBuilder::default();
+        let (mut keys, input) = MintClient::ecash_input(notes).unwrap();
+        builder.input(&mut keys, input);
+        let tx_typed = user.tx_with_change(builder, sats(5000)).await;
+        let tx = tx_typed.into_type_erased();
+        let mint_id = LEGACY_HARDCODED_INSTANCE_ID_MINT;
+
+        // submit the tx successfully
+        let response = fed.submit_transaction(tx.clone()).await;
+        assert_matches!(response, Ok(()));
+        fed.run_empty_epochs(2).await;
+        assert!(fed.find_module_item(mint_id).is_some());
+        fed.clear_ecash_nonces().await;
```

Comment:
Could we verify that it failed at API level by comparing error message instead of clearing spent nonces?

RN if both `fed.clear_ecash_nonces` and replay prevention logic are buggy, then you might still get an error and not notice, so the failure mode is not ideal.

Reply:
I do check for that at the API level, but at the P2P level I'm not sure how to easily get at that error.

Reply:
Oh, OK.